vm/src/stdlib/pystruct.rs

```diff
@@ -72,26 +74,56 @@ mod _struct {
         }
     }
 
-    #[derive(Debug)]
+    #[derive(Debug, Clone)]
     struct FormatSpec {
         endianness: Endianness,
         codes: Vec<FormatCode>,
     }
 
     impl FormatSpec {
+        fn decode_and_parse(
+            vm: &VirtualMachine,
```

Comment:
If you'd like, you could probably use `AsciiStr` from the `ascii` crate to simplify parsing/validating fmt strings.

Reply:
That would be a lot though, probably best left to a separate PR

Reply:
Yeah will do it in a separate PR.